control-plane/csi-driver/src/bin/node/nodeplugin_grpc.rs

```diff
@@ -54,6 +65,73 @@ impl NodePlugin for NodePluginSvc {
             device_path: device.devname(),
         }))
     }
+
+    async fn cleanup(
+        &self,
+        request: Request<CleanupRequest>,
+    ) -> Result<Response<CleanupReply>, Status> {
+        if !self.stale_entry_cleanup {
+            return Ok(Response::new(CleanupReply {}));
+        }
+        let volume_id = request.into_inner().volume_id;
+        debug!("Starting cleanup for volume: {volume_id}");
+
+        let nqn = format!("{}:{volume_id}", nvme_target_nqn_prefix());
+        match Subsystem::try_from_nqn(&nqn) {
+            Ok(subsystem_paths) => {
+                for subsystem_path in subsystem_paths {
```

Comment:
Should we not also unmount the paths? Otherwise it might cause stale mounts?

Reply:
We are unmounting, once the subsystem paths are disconnected, if you see below line 116 and 128.

Reply:
If I am reading correctly, we are unmounting only in case of NqnNotFound, and not after subsystems were found with the nqn and paths disconnected.

Reply:
> If I am reading correctly, we are unmounting only in case of NqnNotFound, and not after subsystems were found with the nqn and paths disconnected.

Yes, that's true, the reason is we cannot unmount if the subsystem path is present as that will get stuck, also we cannot just disconnect and do the umount because the disconnect takes quite a while to remove the path and the path stays in deleting (no IO) in the process of getting removed. For that we issue disconnect and return error so that csi retries and the check happens again, eventually in one of the retries it will be removed and we get into the section of NqnNotFound where we do the unmounts.

Reply:
But we still should unmount the paths nevertheless
You can send unmount lazy after you disconnect?

Reply:
As discussed kept it same for now, would refactor to retry in the method itself.